liferay-headless/liferay-portal
# LPS-165373 Expose sorting functionality to templates

With these changes we can sort results when calling headless resources from templates. 

```
<#assign 
	blogEntriesResource = resourceLocator.locate(
		"/headless-delivery/v1.0", "BlogPosting"
	)
/>

${
	blogEntriesResource.getSiteBlogPostingsPage(
		20121,
		null,
		null,
		blogEntriesResource.toFilter("creatorId eq 20125"),
		null,
		blogEntriesResource.toSorts("dateCreated:asc")
	)
}
```

To test it:

1. Go to `Design -> Templates -> Widget Templates`
2. Create a `Blogs Template`.
3. Add the Blogs Widget to the page.
4. In the configuration of the widget, under `Display Template`, select the template created in step 2.
5. Create a blog entry and copy its id.
6. Edit the template from step 2 and add this:
```
<#assign 
	blogEntriesResource = resourceLocator.locate(
		"/headless-delivery/v1.0", "BlogPosting"
	)
/>

${
	blogEntriesResource.getSiteBlogPostingsPage(
		20121,
		null,
		null,
		blogEntriesResource.toFilter("creatorId eq 20125"),
		null,
		blogEntriesResource.toSorts("dateCreated:asc")
	)
}
```
7. Render the homepage and you should see the json with the entries sorted by dateCreated.
